src/util/modules.py

```diff
@@ -156,9 +161,9 @@ def uninstall_module(cr, module):
                                   AND module != d.module)
                AND module = %s
                AND model != 'ir.module.module'
-          ORDER BY id DESC
+          ORDER BY array_position(%s, model) NULLS LAST, array_position(%s, model) NULLS FIRST, id DESC
```

Comment:
We at least need the `model::text` but it helps to be explicit about the array type also...
```suggestion
          ORDER BY array_position(%s::text[], model::text) NULLS LAST, array_position(%s::text[], model::text) NULLS FIRST, id DESC
```

Reply:
Are you sure it is required? It never failed during my tests.

Reply:
Well it fails in matt:
```
  File "/upgrade-util/src/util/modules.py", line 152, in uninstall_module
    cr.execute(
  File "/tmp/tmpmn5jisaa/odoo/master/odoo/sql_db.py", line 354, in execute
    res = self._obj.execute(query, params)
psycopg2.errors.UndefinedFunction: function array_position(text[], character varying) does not exist
LINE 12:           ORDER BY array_position(ARRAY['ir.rule'], model) N...
                            ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
```

Reply:
It looks like it works from PG14 up :/ (PG version strikes again)
```
psql (16.4 (Ubuntu 16.4-1.pgdg20.04+1), server 14.13 (Ubuntu 14.13-1.pgdg20.04+1))
Type "help" for help.

afu_test=> select ARRAY_POSITION(array['base'], 'base'::varchar);
 array_position
----------------
              1
```
while PG13:
```
psql (16.4 (Ubuntu 16.4-1.pgdg20.04+1), server 13.16 (Ubuntu 13.16-1.pgdg20.04+1))
Type "help" for help.

afu_test=> select ARRAY_POSITION(array['base'], 'base'::varchar);
ERROR:  function array_position(text[], character varying) does not exist
LINE 1: select ARRAY_POSITION(array['base'], 'base'::varchar);
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
```

Reply:
If only I had look at the failed CI... 🤦